sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -4882,3 +4882,65 @@ def sdc_pandas_series_skew_impl(self, axis=None, skipna=None, level=None, numeri
         return numpy_like.skew(self._data)
 
     return sdc_pandas_series_skew_impl
+
+
+@sdc_overload_method(SeriesType, 'combine')
+def sdc_pandas_series_combine(self, other, func, fill_value=None):
+    """
+    Intel Scalable Dataframe Compiler User Guide
+    ********************************************
+
+    Pandas API: pandas.Series.combine
+
+    Limitations
+    -----------
+    - Only supports the case when data in series of the same type
+
+    Examples
+    --------
+    .. literalinclude:: ../../../examples/series/series_combine.py
+       :language: python
+       :lines: 27-
+       :caption: Combined the Series with a Series according to func.
+       :name: ex_series_combine
+
+    .. command-output:: python ./series/series_combine.py
+       :cwd: ../../../examples
+
+    Intel Scalable Dataframe Compiler Developer Guide
+    *************************************************
+    Pandas Series method :meth:`pandas.Series.combine` implementation.
+
+    .. only:: developer
+
+       Tests: python -m sdc.runtests -k sdc.tests.test_series.TestSeries.test_series_combine*
+    """
+    _func_name = 'Method Series.combine().'
+
+    ty_checker = TypeChecker(_func_name)
+    ty_checker.check(self, SeriesType)
+
+    ty_checker.check(other, SeriesType)
+
+    if not isinstance(fill_value, (types.Omitted, types.NoneType, types.Number)) and fill_value is not None:
+        ty_checker.raise_exc(fill_value, 'number', 'fill_value')
+
+    def sdc_pandas_series_combine_impl(self, other, func, fill_value=None):
+
+        if fill_value is None:
+            fill_value = numpy.nan
+
+        len_val = max(len(self), len(other))
+        result = numpy.empty(len_val, self._data.dtype)
```

Comment:
This is actually wrong, result dtype should be common dtype for result dtype of func(a, b) where a,b are series values and dtype of _fill_value. Provided tests do not cover this, but e.g. this won't pass:
```
    def test_series_combine_integer_new(self):
        def test_impl(S1, S2):
            return S1.combine(S2, lambda a, b: 2 * a + b, 16.2)
        hpat_func = self.jit(test_impl)

        S1 = pd.Series([1, 2, 3, 4, 5])
        S2 = pd.Series([6, 21, 3, 5])
        result = hpat_func(S1, S2)
        result_ref = test_impl(S1, S2)
        print(f"DEBUG: result:\n{result},\nresult_ref:\n{result_ref}")
        pd.testing.assert_series_equal(result, result_ref)
```